src/jit/codegenxarch.cpp

```diff
@@ -718,7 +718,7 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
         // Note that a reg optional operand is a treated as a memory op
         // if no register is allocated to it.
         assert(divisor->isUsedFromReg() || divisor->isMemoryOp() || divisor->IsCnsFltOrDbl() ||
-               divisor->IsRegOptional());
+               divisor->IsRegOptional() || divisor->IsLocal());
```

Comment:
I believe that this is indeed exhaustive. If it has a register, the first condition will be true. Otherwise, if it is a legal operand to divide, it will be one of:
- a spilled tree temp (in which case `isMemoryOp()` will be true
- an indirection or `GT_CLS_VAR` (in which case `isMemoryOp()` will be true
- a `RegOptional()` memory operand
- a local

I can't think of any other case - so I think this is a useless assert.